doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,470 @@
+
+.. rubric:: Returns
+
+A ciphertext as a binary string.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the encryption algorithm supports `RSA` to encrypt the string.
+
+* str - measured in bytes. The length of the string must not be greater than the key_str modulus length in bytes - 11 (additional bytes used for PKCS1 padding)
+
+* key_str - a key (either private or public) in the PEM format
+
+.. _asym_sign:
+
+asymmetric_sign(*algorithm, digest_str, priv_key_str, digest_type*)
+------------------------------------------------------------------------
+
+Signs a digest string using a private key string.
+
+.. rubric:: Returns
+
+A signature is a binary string.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the encryption algorithm supports either `RSA` or `DSA` to encrypt the string.
+
+* digest_str - the digest binary string that is signed. Invoking :ref:`create_digest<create_digest>` generates the string.
+
+* priv_key_str - the private key used to sign the digest string. The key must be in the PEM format.
+
+* digest_type - the supported values are listed in the digest type table of :ref:`create_digest<create_digest>`.
+
+.. _asym_verify:
+
+asymmetric_verify(*algorithm, digest_str, sig_str, pub_key_str, digest_type*)
+--------------------------------------------------------------------------------
+
+Verifies whether the signature string matches the digest string.
+
+.. rubric:: Returns
+
+A ``1`` (success) or a ``0`` (failure).
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - supports either 'RSA' or 'DSA'.
+
+* digest_str - invoking :ref:`create_digest<create_digest>` generates the digest binary string and this string.
+
+* sig_str - the signature binary string. Invoking :ref:`asymmetric_sign<asym_sign>` generates this string.
+
+* pub_key_str - the signer's public key string. This string must correspond to the private key passed to :ref:`asymmetric_sign<asym_sign>` to generate the signature string. The string must be in the PEM format.
+
+* digest_type - the supported values are listed in the digest type table of :ref:`create_digest<create_digest>`
+
+.. _create_priv_key:
+
+create_asymmetric_priv_key(*algorithm, (key_len | dh_parameters)*)
+--------------------------------------------------------------------------------
+
+Generates a private key using the given algorithm and key length for RSA or DSA or Diffie-Hellman parameters (for DH).
+
+.. rubric:: Returns
+
+The key as a string in PEM format.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA', 'DSA', or 'DH'.
+
+* key_len - the supported key length values are the following:
+
+  - RSA - the minimum length is 1,024. The maximum length is 16,384. The default value is 16,384.
+
+  - DSA - the minimum length is 1,024. The maximum length is 9,984. The default value is 9,984.
```

Comment:
The function does not have default parameters.
```suggestion
  - DSA - the minimum length is 1,024. The maximum length is 9,984.
```

Reply:
DSA section - removed the default.